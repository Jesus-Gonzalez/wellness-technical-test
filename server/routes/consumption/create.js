import Models from '~/database/models'

export const consumptionPostRoute = app => {
  app.post('/consumption', (req, res, next) => {
    const consumption = req.body

    if ((
      !consumption ||
      !consumption.consumption ||
      !consumption.cost ||
      !consumption.price ||
      !consumption.date
    )) {
      res.status(400).end('request-body-empty')
    }

    const model = new Models.ConsumptionModel(req.body)

    return model.save()
      .then(consumption => {
        console.log('consumption', consumption)
        res.status(200).end(JSON.stringify(consumption))
      })
      .catch(() => {
        res.status(500).end('saving')
      })
      .finally(() => {
        next()
      })
  })
}
