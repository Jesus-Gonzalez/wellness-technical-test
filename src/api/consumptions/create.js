import { HttpClient } from 'api/http'

export const createConsumption = data => (
  HttpClient({
    url: '/consumption',
    method: 'post',
    data
  }).then(({ data }) => data)
)
