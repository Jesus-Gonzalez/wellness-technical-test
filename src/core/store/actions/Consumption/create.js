import * as ActionTypes from 'core/store/actionTypes'

export const createConsumption = item => ({
  type: ActionTypes.Consumption.Create,
  item
})
