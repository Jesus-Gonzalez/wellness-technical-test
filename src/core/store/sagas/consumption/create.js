import { call, put, takeLatest } from 'redux-saga/effects'

import * as ActionTypes from 'core/store/actionTypes'
import * as Api from 'api'

function* createConsumption(action) {
  const { item } = action
  try {
    const consumption = yield call(Api.createConsumption, item)
    yield put({ type: ActionTypes.Consumption.CreateSuccess, item: consumption })
  } catch (error) {
    yield put({ type: ActionTypes.Consumption.CreateError, item })
  }
}

export default function* updateConsumptionSaga() {
  yield takeLatest(ActionTypes.Consumption.Create, createConsumption)
}
