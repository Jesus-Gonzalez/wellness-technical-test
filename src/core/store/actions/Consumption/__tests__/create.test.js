import { createConsumption } from '../create'

describe('redux', () => {
  describe('actions', () => {
    describe('create', () => {
      it('should return an action', () => {
        expect(createConsumption({ test: 'test' })).toMatchSnapshot()
      })
    })
  })
})
