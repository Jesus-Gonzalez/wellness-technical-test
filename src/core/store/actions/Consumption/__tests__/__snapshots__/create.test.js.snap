// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux actions create should return an action 1`] = `
Object {
  "item": Object {
    "test": "test",
  },
  "type": "CREATE_CONSUMPTIONS",
}
`;
